xwiki-platform-core/xwiki-platform-web/src/main/webapp/resources/js/xwiki/table/livetabledate.js

```diff
@@ -0,0 +1,93 @@
+require.config({
+  paths: {
+    'moment': "$services.webjars.url('momentjs', 'moment.js')",
+    'jdateformatparser': "$services.webjars.url('org.webjars.bower:moment-jdateformatparser', 'moment-jdateformatparser.js')",
+    'daterangepicker': "$services.webjars.url('bootstrap-daterangepicker', 'js/bootstrap-daterangepicker.js')"
+  },
+  shim: {
+    'moment': ['jdateformatparser'],
+    'daterangepicker': ['jquery', 'bootstrap', 'moment']
+  }
+});
+
+require(['jquery', 'moment', 'daterangepicker'],
+  function($, moment) {
+    var dateFormat = moment().toMomentFormatString('$escapetool.javascript($xwiki.getXWikiPreference("dateformat", "yyyy/MM/dd HH:mm"))');
```

Comment:
I would rather read the date format from an attribute of the input field that is enhanced by the date range picker. This means calling ``getXWikiPreference()`` in ``macros.vm``. Moreover, Date xproperties can overwrite the date format specified in XWiki preferences and it's easier to get the date format from the xclass in ``macros.vm`` rather than in JavaScript.